Send new review number two

MasakiShiraishi opened the pull request:
# Description

I created a format for submitting names, comments, and ratings.

# Screenshots (if needed)
![2024-02-05 (2)](https://github.com/JrSchoolAccount/Group-Assignment-Cinema-Project-2.0/assets/112933577/05aceed2-097c-4948-8993-df6b26cef2d6)
![2024-02-09 (2)](https://github.com/JrSchoolAccount/Group-Assignment-Cinema-Project-2.0/assets/112933577/963eda60-e702-4a46-a9d0-8ffc34db32a4)
![2024-02-09 (4)](https://github.com/JrSchoolAccount/Group-Assignment-Cinema-Project-2.0/assets/112933577/a5c3f2a1-9e09-4ea6-b5f6-e3a2bf1ca31b)
![2024-02-09 (3)](https://github.com/JrSchoolAccount/Group-Assignment-Cinema-Project-2.0/assets/112933577/06f39b14-d3e1-4a58-bcdf-0ec108020d13)

MasakiShiraishi committed:
adding sass:watch in package.json

MasakiShiraishi committed:
adjusting to the sass file

MasakiShiraishi committed:
adding review form

MasakiShiraishi committed:
installed render function

MasakiShiraishi committed:
adding style to review form

MasakiShiraishi committed:
handling merge conflict

NelsonTheBrave reviewed (commented):
Looks nice overall masaki! Just wondering about this part, is it the same code two times or am I missing something?

ALSO! It seems you haven't applied the mutual settings for Prettier since you have double quotation instead of single quotation like everyone else. I think you probably should change those settings before we merge

Also there a some minor merge conflicts that are super easily fixed, might have come since I merged Felix's PR.

![image](https://github.com/JrSchoolAccount/Group-Assignment-Cinema-Project-2.0/assets/145323995/aceae136-ebd5-4889-af36-a9d7874407c0)

MasakiShiraishi committed:
removing code duplication

MasakiShiraishi commented:
Sorry about that!
I made a mistake while handling a merge conflict, but I've fixed it.


![2024-02-09 (6)](https://github.com/JrSchoolAccount/Group-Assignment-Cinema-Project-2.0/assets/112933577/a3693200-b2af-4a99-9a7f-ef6001a9e0e5)

MasakiShiraishi committed:
handling merge conflict agin

MasakiShiraishi committed:
handling merge conflict

JrSchoolAccount commented on src/movies.js:
Looks good 👍 
Maybe consider moving the code to a separate file or implement the cmsAdapter 😊

MasakiShiraishi replied:
I created one more new review.js file.

JrSchoolAccount reviewed (dismissed):
👍

MasakiShiraishi committed:
creating reviews.js

NelsonTheBrave reviewed (commented):
Good, I think you made good changes! Seems like there's a minor merge conflict to fix, then ready to merge I think!

MasakiShiraishi committed:
handling merge conflict

MasakiShiraishi commented:
I handled merge conflicts again. Could you check again, please?

NelsonTheBrave reviewed (dismissed):
Thumbs up!

FelixHultman reviewed (dismissed):
Good work Masaki, looking very good

MasakiShiraishi committed:
merge from main

MasakiShiraishi committed:
removing unnecessary code and changing to single quote

MasakiShiraishi commented:
Hi,@JrSchoolAccount
I removed unnecessary code from reviews.js & app.js and also changed double quotes to single quotes.
![2024-02-10 (1)](https://github.com/JrSchoolAccount/Group-Assignment-Cinema-Project-2.0/assets/112933577/4ff18ab1-a46a-4ad1-9e82-1db8931a65bc)
![2024-02-10](https://github.com/JrSchoolAccount/Group-Assignment-Cinema-Project-2.0/assets/112933577/7c7f60a1-6559-43b5-abb3-71c0bc304567)

JrSchoolAccount reviewed (commented):
Looks like you misinterpreted the assignment.
The frontend code with fetch() is missing from this pull request.
Im a wrong or are you planing to do the frontend code at a later point?

Other than that the structure of the code is really good and if the post was to be handled with the backend it would have been really good 😁

NelsonTheBrave reviewed (changes requested):
Yes, after Jesper's comment that there seems to be some misunderstanding of the assignment I also realise that there are still changes that need to be made, that I didn't see when I looked through it before (obviously too fast :O ). The assignment states clearly that our API should be fetched using fetch() as Jesper pasted in. You seem to be using a post method on a submit form in you reviews.ejs, which I guess does something similar to fetch(), but also as Jesper said we have to remember it should be made frontend according to the instructions :O

MasakiShiraishi commented:
I've been trying to redo it as you suggested since yesterday, but it hasn't worked.
It looks like it will take some time.

MasakiShiraishi committed:
deleted

MasakiShiraishi committed:
created för fetch data

MasakiShiraishi committed:
adjusting

MasakiShiraishi committed:
created for sending data till JSON file

MasakiShiraishi commented:
I corrected the errors that I had made. I hope I'm not wrong this time.

NelsonTheBrave commented on src/app.js:
```diff
@@ -49,6 +54,42 @@ app.get('/filmer/:movieId', async (req, res) => {
+  res.render('reviews', { movieId });
+});
+
+app.post('/api/movies/:movieId/reviews', (req, res) => {
```
Is there a way you could put most of the content of this app.post into a separate js file, where most of the functions and operations happen? I think that is the way we have done with the other routes to keep the app.js a bit cleaner and more easily overlooked. Maybe you can put it in the reviewBody.js that you already import?

NelsonTheBrave reviewed (changes requested):
I think you are starting to get to a better solution. From what I can tell you are now frontend fetching from our API the way it should be done according to instructions. You still have several console.logs that you haven't removed from your code, so please take a look at that as well!

JrSchoolAccount reviewed (commented):
Much better, but have in mind separation of concerns. 
Like Erik said, re structure the code and put it in a separate file, så that app.js is mostly for the routes.

MasakiShiraishi committed:
adjusting code

MasakiShiraishi commented:
I attempted to move the post method to reviewBody.js as suggested by Erik and Jesper. I think that the program seems to be working as expected.

![2024-02-14 (1)](https://github.com/JrSchoolAccount/Group-Assignment-Cinema-Project-2.0/assets/112933577/28e61b2a-e864-42f1-87d0-fe281adbc36c)
![2024-02-14](https://github.com/JrSchoolAccount/Group-Assignment-Cinema-Project-2.0/assets/112933577/124f394b-3a27-4a81-a5eb-2d5096f9b2f8)
![2024-02-14 (3)](https://github.com/JrSchoolAccount/Group-Assignment-Cinema-Project-2.0/assets/112933577/df82ca74-a74d-4a60-8c86-6156239f68c0)

simonpizevski reviewed (commented):
Looks good, the seperation of the code looks good too! Great job!

richardolsson commented on src/app.js:
```diff
@@ -49,6 +53,24 @@ app.get('/filmer/:movieId', async (req, res) => {
+    rating,
+    author,
+  };
+  await review(newReview);
```
I find the name of this function to be very ambiguous. Because I know the assignment, I can guess what it does, but I would have still used a more explicit name, like `sendReview()`, `submitReview()` or something like that.

And the best thing of course would be to make it a method on an API adapter. That way, it can more easily be mocked and tested.

richardolsson commented on src/app.js:
```diff
@@ -7,8 +7,12 @@ import { loadMovieReviews } from './movieReviews.js';
+import { review } from './reviewBody.js';
 
 const app = express();
+app.use(express.urlencoded({ extended: true }));
```
I don't think this is used anymore. If I'm correct, it can be deleted. It's always good to get rid of "dead code" right away, because the more time passes the more difficult it's going to be to remember if it's really dead or why it was ever there.

richardolsson commented on src/reviewBody.js:
```diff
@@ -0,0 +1,47 @@
+  const date = new Date().toISOString();
+  const body = {
+    data: {
+      comment: 'string',
+      rating: 0,
+      author: 'string',
+      verified: true,
+      movie: 'string or id',
+      createdAt: date,
+      updatedAt: date,
+      createdBy: 'string or id',
+      updatedBy: 'string or id',
```
This doesn't look right to me. Is this really working?

My guess is that you've read the API documentation and found that `comment` should be a string, but that doesn't mean that it should have the value `"string"` (that specific piece of text) but that it should have _some_ piece of text.

Because of `...newReview` on the next line, and the fact that `newReview` will contain `movie`, `comment`, `rating` and `author`, those very important fields will have their values overwritten by the real ones. But you still have `'string or id'` (that exact piece of text) as the value for `createdBy` and `updatedBy` and that's not correct. I'm surprised that it even works – I would have expected the CMS to complain about them.

richardolsson reviewed (commented):
Great work on this! I have left some comments below that you can hopefully learn something from for future assignments.
